march_rqt_input_device/src/march_rqt_input_device/input_device_view.py

```diff
@@ -20,18 +22,32 @@ def __init__(self, ui_file, controller):
         """
         super(InputDeviceView, self).__init__()
         self._controller = controller
+        self._controller.accepted_cb = self._accepted_cb
+        self._controller.finished_cb = self._finished_cb
+        self._controller.rejected_cb = self._rejected_cb
+        self._controller.current_gait_cb = self._current_gait_cb
+
+        self._always_enabled_buttons = []
 
         # Extend the widget with all attributes and children from UI file
         loadUi(ui_file, self)
 
+        self.refresh_button.clicked.connect(self._update_possible_gaits)
+
+        self._create_buttons()
+        self._update_possible_gaits()
+
+    def _create_buttons(self):
         # Create buttons here
         rocker_switch_increment = \
             self.create_button('rocker_switch_up', image_path='/rocker_switch_up.png',
-                               callback=lambda: self._controller.publish_increment_step_size())
+                               callback=lambda: self._controller.publish_increment_step_size(),
+                               always_enabled=True)
 
         rocker_switch_decrement = \
             self.create_button('rocker_switch_down', image_path='/rocker_switch_down.png',
-                               callback=lambda: self._controller.publish_decrement_step_size())
+                               callback=lambda: self._controller.publish_decrement_step_size(),
+                               always_enabled=True)
```

Comment:
Why are these always enabled? I think they should only be enabled during walk at the moment.

Reply:
I have made these always enabled, because currently it is not possible to request from the statemachine if it is possible to increment or decrement step size. I could check for walking, since it only works during walk, but it might be nicer to create an interface to the state machine for this. Then you can also add cases where you can only increment or decrement.